### ISSUE TITLE
Translation providers merging is not always done in the correct order.

### ISSUE THREAD

**Issue by dsuwirya** [robisim74/angular-l10n#182]
Hello,

Let's say that the current language is set to fr-FR and translations providers stack looks like the following:

translation: {
providers: [
...
{ type: ProviderType.WebAPI, path: 'https://www.myserver.com/i18n/en.json' },
{ type: ProviderType.WebAPI, path: 'https://www.myserver.com/i18n/fr.json' },
{ type: ProviderType.WebAPI, path: 'https://www.myserver.com/i18n/fr-FR.json' },
]}

When the language json files got downloaded in the right order by the browser ( e.g. en, fr, fr-FR ), then the merging is done correctly ( i.e. en > fr > fr-FR ). 

However, if for any reason json files got downloaded not in the right order ( e.g. fr, fr-FR, en ), then merging order is becoming wrong ( i.e. fr > fr-FR > en ), hence displayed text will be in English.

Would it be possible to make sure that the merging is always correctly done ( as per the translation providers stack order ), irrespective of the json files download order ?

Thanks.

**Comment by robisim74** [comment]
The merging is in order only if you use _providerType.fallback_, but by default it supports only static json: https://robisim74.github.io/angular-l10n/spec/configuration/#using-fallback-providers.
However you could implement a custom provider to change the default behavior: https://robisim74.github.io/angular-l10n/spec/configuration/#using-a-custom-provider

The default behavior: https://github.com/robisim74/angular-l10n/blob/master/src/services/translation-provider.ts#L37

**Comment by dsuwirya** [comment]
Can't you do something in TranslationService > getTranslation(), to first accumulate the incoming json files in the proper order, forkJoin, then invoke the merge operation only after all downloads completed ? 

Thanks.

**Comment by robisim74** [comment]
As I said, that's already done for fallback providers (and you have fallback providers). Only difference is that you have also web api: because the cascade order is slower than the merging, by default are used json files, and not web api (usually still slower, also if you are setting a different timeout).

You should change only here: https://github.com/robisim74/angular-l10n/blob/master/src/services/translation-provider.ts#L37

**Comment by dsuwirya** [comment]
Currently, this is how my full language providers stack look like:

translation: {
providers: [
{ type: ProviderType.Fallback, prefix: './assets/i18n/locale-en.json' },
{ type: ProviderType.Fallback, prefix: './assets/i18n/locale-fr.json'},
{ type: ProviderType.Fallback, prefix: './assets/i18n/locale-fr-FR.json' },
{ type: ProviderType.WebAPI, path: 'https://www.myserver.com/i18n/en.json' },
{ type: ProviderType.WebAPI, path: 'https://www.myserver.com/i18n/fr.json' },
{ type: ProviderType.WebAPI, path: 'https://www.myserver.com/i18n/fr-FR.json' }
]}

So are u saying that I should be categorizing all of them as ProviderType.Fallback in order to be sure that they will be merged always in the correct order? Plus of course, implement the custom provider to be able to fetch the 'Fallback' type from remote as well?

Thanks.

**Comment by robisim74** [comment]
Yes.

**Comment by dsuwirya** [comment]
Thanks much! I did what you suggested and seems to always merge correctly now.
